src/Features/Core/Portable/Formatting/FormattingCodeFixProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Formatting
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = PredefinedCodeFixProviderNames.FixFormatting)]
+    [Shared]
+    internal class FormattingCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.FormattingDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(c => FixOneAsync(context.Document, context.Diagnostics, c)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected async Task<Document> FixOneAsync(Document document, ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var changes = new List<TextChange>();
+            foreach (var diagnostic in diagnostics)
+            {
+                if (!tree.Equals(diagnostic.Location.SourceTree)
+                    || !diagnostic.Properties.TryGetValue(FormattingDiagnosticAnalyzer.ReplaceTextKey, out var replacement))
+                {
+                    continue;
+                }
+
+                changes.Add(new TextChange(diagnostic.Location.SourceSpan, replacement));
+            }
+
+            changes.Sort((left, right) => left.Span.Start.CompareTo(right.Span.Start));
```

Comment:
is WithChanges expect changes to be ordered? is that why we sort it here? or WithChanges does the sorting already? what happens if start location is same?

Reply:
or has intersected or overlapped changes?

Reply:
> is WithChanges expect changes to be ordered?

It now handles either case, but is more efficient when the input is ordered.

> what happens if start location is same?

This is valid only for the case of multiple insertions in the same location, a condition which is not expected to occur for this code fix.

> or has intersected or overlapped changes?

This isn't valid regardless of order.

Reply:
The sorting operation was removed